openapi/access.yaml

```diff
@@ -483,31 +471,25 @@ components:
         - KMAC128
     Collection:
       type: object
-      required:
-        - id
-        - transactions
       properties:
         id:
           $ref: '#/components/schemas/Identifier'
         transactions:
           type: array
           items:
             $ref: '#/components/schemas/Transaction'
+        _expandable:
```

Comment:
Without `expand=transaction`

> {
>   "id": "string",
>   "_expandable": [
>     {
>       "transaction": "/v1/transactions/1234123124"
>     },
>     {
>       "transaction": "/v1/transactions/1234123125"
>     },
>     {
>       "transaction": "/v1/transactions/1234123126"
>     },
>   ],
>   "_links": {
>     "_self": "string"
>   }
> }

Reply:
With `expand=transaction`

> {
>   "id": "string",
>   "transactions": [
>     {
>       "id": "string",
>       "script": "string",
>       "arguments": [
>         "string"
>       ],
>       "reference_block_id": "string",
>       "gas_limit": 0,
>       "payer": "string",
>       "proposal_key": {
>         "address": "string",
>         "key_index": 0,
>         "sequence_number": 0
>       },
>       "authorizers": [
>         "string"
>       ],
>       "payload_signatures": [
>         {
>           "address": "string",
>           "signer_index": 0,
>           "key_index": 0,
>           "signature": "string"
>         }
>       ],
>       "envelope_signatures": [
>         {
>           "address": "string",
>           "signer_index": 0,
>           "key_index": 0,
>           "signature": "string"
>         }
>       ],
>       "result": {
>         "block_id": "string",
>         "status": "Pending",
>         "error_message": "string",
>         "computation_used": 0,
>         "events": [
>           {
>             "type": "string",
>             "transaction_id": "string",
>             "transaction_index": 0,
>             "event_index": 0,
>             "payload": "string"
>           }
>         ],
>         "_links": {
>           "_self": "string"
>         }
>       },
>       "_expandable": {
>         "result": "string"
>       },
>       "_links": {
>         "_self": "string"
>       }
>     },
>     {
>       "id": "string",
>       "script": "string",
>       "arguments": [
>         "string"
>       ],
>       "reference_block_id": "string",
>       "gas_limit": 0,
>       "payer": "string",
>       "proposal_key": {
>         "address": "string",
>         "key_index": 0,
>         "sequence_number": 0
>       },
>       "authorizers": [
>         "string"
>       ],
>       "payload_signatures": [
>         {
>           "address": "string",
>           "signer_index": 0,
>           "key_index": 0,
>           "signature": "string"
>         }
>       ],
>       "envelope_signatures": [
>         {
>           "address": "string",
>           "signer_index": 0,
>           "key_index": 0,
>           "signature": "string"
>         }
>       ],
>       "result": {
>         "block_id": "string",
>         "status": "Pending",
>         "error_message": "string",
>         "computation_used": 0,
>         "events": [
>           {
>             "type": "string",
>             "transaction_id": "string",
>             "transaction_index": 0,
>             "event_index": 0,
>             "payload": "string"
>           }
>         ],
>         "_links": {
>           "_self": "string"
>         }
>       },
>       "_expandable": {
>         "result": "string"
>       },
>       "_links": {
>         "_self": "string"
>       }
>     }
>   ],
>   "_links": {
>     "_self": "string"
>   }
> }

Reply:
Without `expand=transactions`

```
{
    "id": "string",
    "_expandable": {
        "transactions": [
            "/v1/transactions/1234123124",
            "/v1/transactions/1234123125",
            "/v1/transactions/1234123126"
        ]
    },
    "_links": {
        "_self": "string"
    }
}
```

Reply:
We can keep `_expandable` as an object rather than a list (there may be other fields than are added to `_expandable`, so I don't want to turn it into a list just for this specific API)

Reply:
So it we decide to go forward with this change, then the convention for `_expandable` going forward will be as follows:
* If the expandable key is a sub-resource, the value should be the link to the resource
* If the expandable key is a regular field, the value should be empty string
* If the expandable key is a list of sub-resources, the value should be a list of links

Reply:
Having said all of that, in the end I still tend to prefer approach 1: https://github.com/onflow/flow/pull/705#issuecomment-983943265

Reply:
I like the this solution and have made changes for it. My reservation against approach 1 is that we are relying on optional query params to safe-gaurd against unintentional performance hit.

For point #2 - `If the expandable key is a regular field, the value should be empty string`, I was thinking of adding default values to the expandable as `""`. This helps a little in the implementation where we can stick in the default generated model for the expandable. Will create another PR for it